shopfloor/services/zone_picking.py

```diff
@@ -510,17 +524,20 @@ def _scan_source_location(
             message = self.msg_store.location_not_allowed()
             return response, message
 
-        product, lot, package = self._find_product_in_location(location, product_id)
+        product, lot, packages = self._find_product_in_location(
```

Comment:
If we change location, we have to remove the package from the context
```suggestion
        if package and package.location_id != location:
          package = False
        product, lot, packages = self._find_product_in_location(
```

Reply:
Fixed